nexus/src/app/sagas/region_snapshot_replacement_start.rs

```diff
@@ -1534,17 +1534,37 @@ pub(crate) mod test {
         request: &RegionSnapshotReplacement,
     ) {
         let opctx = test_opctx(cptestctx);
-        let db_request = datastore
-            .get_region_snapshot_replacement_request_by_id(&opctx, request.id)
-            .await
-            .unwrap();
 
-        assert_eq!(db_request.new_region_id, None);
-        assert_eq!(
-            db_request.replacement_state,
-            RegionSnapshotReplacementState::Requested
-        );
-        assert_eq!(db_request.operating_saga_id, None);
+        let mut i = 1;
+        loop {
+            let db_request = datastore
+                .get_region_snapshot_replacement_request_by_id(
+                    &opctx, request.id,
+                )
+                .await
+                .unwrap();
+
+            assert_eq!(db_request.new_region_id, None);
+            assert_eq!(db_request.operating_saga_id, None);
+
+            if !matches!(
+                db_request.replacement_state,
+                RegionSnapshotReplacementState::Requested
+            ) {
+                eprintln!(
+                    "loop {i} Failed {:?} != Requested",
```

Comment:
What state was this in when you saw this? This seems like a failure to properly unwind the saga as it should never have reached this check without going through `rsrss_set_saga_id_undo`. Likewise, the `assert_eq` above checking `operating_saga_id` should have fired if the request was in an intermediate state.

Reply:
It does eventually get to the requested state.
When I saw the panic, it would be in Allocating state.  At least that's what I had written down in an earlier comment.

Reply:
As discussed in DMs, this was concerning enough to address in a separate PR.